### PR TITLE
Fail safe makedirs

### PR DESCRIPTION
**Pull request by imneonizer**
A more pythonic way to create directories if doesn't exists.

### CODE REVIEW

**Comment by roclark** [comment]
Hey @imneonizer, thanks for putting this together! This looks great and I'd be happy to merge it, but do you mind putting a "Signed-Off-By: Name _email_" at the end of your commit? Something like:

```
Fail safe makedirs

A more Pythonic ...

Signed-Off-By: First Last <email@domain.com>
```

This just helps us ensure all contributors acknowledge the Contributors License Agreement for the repo. Once that's done and the tests pass, I'd be happy to merge!

**Comment by imneonizer** [comment]
@roclark Thank you for the info, I have updated the commit message, hope this works.

**Comment by roclark** [comment]
Thanks for following up @imneonizer! I believe I may have misspoke earlier - I meant to say the commit message I used as an example above should be embedded in your actual commit as opposed to the pull request on GitHub. That way, your `Signed-Off-By: ...` can be seen when someone does a `git log` to view all commits/messages from a command line. This can be done with a `git commit --amend` or similar to update the existing commit, followed by a `git push --force`.

Also, it appears the CI tests are currently failing with an extra whitespace. I can point this out in the code if that helps. Preferably, the updated commit message and the removal of the extra whitespace will be done in the same, single commit.

Thanks again for taking the time for this! Please let me know if there are any further questions.

File: imagine/imagine.py [hunk 1]
```diff
@@ -119,8 +119,8 @@ def _try_create_directory(directory: str) -> NoReturn:
     directory : string
         A ``string`` of a path pointing to a directory to attempt to create.
     """
-    if not os.path.exists(directory):
-        os.mkdir(directory)
+
```
Comment (roclark) [inline imagine/imagine.py:122]: This extra blank line is causing the CI tests to fail. Removing this line should clean that up.

**Review by roclark — CHANGES_REQUESTED** [review]

**Comment by imneonizer** [comment]
@roclark
Thanks for such a detailed explanation, I have made the changes.  
This is my first PR in open-source projects at Nvidia, really excited to see it merged!!

**Review by roclark — APPROVED** [review]

**Comment by roclark** [comment]
Hey @imneonizer, sorry for not getting to this yesterday! Thanks for cleaning things up here - this looks good to me and I am happy to merge it! I will do a squash commit to combine your two commits into one and include it with the main branch. Thanks for creating this and I'm happy to play my little part in your first PR with NVIDIA! 😃 I welcome any future PRs you may have as well!
